docs/tutorials/spec2006-tutorial-details.md

```diff
@@ -0,0 +1,636 @@
+We are only interested in the first part of stats.txt.
+
+
+## Documenting the Preparing Steps
+### Setting up the Experiment Folder
+We set up a folder to contain (almost) all materials of the experiment.
+We use git to keep track of changes in the folder.
+
+```sh
+mkdir spec2006-experiments
+cd spec2006-experiments
+git init
+```
+
+We need add a remote to the repository.
+
+```sh
+git remote add origin https://your-remote-add/spec-experiment.git
+```
+
+We document the root folder of the experiment in launch_spec2006_experiments.py as follows,
+
+```sh
+experiments_repo = Artifact.registerArtifact(
+    command = '',
+    typ = 'git repo',
+    name = 'experiment',
+    path =  './',
+    cwd = './',
+    documentation = 'local repo to run spec 2006 experiments with gem5 full system mode'
+)
+```
+
+We use .gitignore file to ignore changes of certain files or folders.
+In this experiment, we will use this .gitignore file,
+
+```
+*.pyc
+m5out
+.vscode
+results
+gem5art-env
+disk-image/packer
+disk-image/spec-2006/spec-2006-image/spec-2006
+disk-image/packer_cache
+disk-image/spec-2006/CPU2006v1.0.1.iso
+gem5
+linux-4.19.83/
+```
+
+Essentially, we will ignore files and folders that we use gem5art to keep track of them, or the presence of those files and folders do not affect the experiment's results.
+
+### Building gem5
+In this step, we download the source code and build gem5 v20.1.0.2.
+
+```sh
+git clone -b v20.1.0.2 https://gem5.googlesource.com/public/gem5
+cd gem5
+scons build/X86/gem5.opt -j8
+```
+
+We have two artifacts: one is the gem5 source code (the gem5 git repo), and the gem5 binary (`gem5.opt`).
+The documentation of this step would be how we get the source code and how we compile the gem5 binary.
+In launch_spec2006_experiments.py, we document the step in Artifact objects as follows,
+
+```python
+gem5_repo = Artifact.registerArtifact(
+    command = '''
+        git clone -b v20.1.0.2 https://gem5.googlesource.com/public/gem5
+        cd gem5
+        scons build/X86/gem5.opt -j8
+    ''',
+    typ = 'git repo',
+    name = 'gem5',
+    path =  'gem5/',
+    cwd = './',
+    documentation = 'cloned gem5 v20.1.0.2'
+)
+
+
+gem5_binary = Artifact.registerArtifact(
+    command = 'scons build/X86/gem5.opt -j8',
+    typ = 'gem5 binary',
+    name = 'gem5-20.1.0.2',
+    cwd = 'gem5/',
+    path =  'gem5/build/X86/gem5.opt',
+    inputs = [gem5_repo,],
+    documentation = 'compiled gem5 v20.1.0.2 binary'
+)
+```
+
+### Building m5
+m5 is a binary that facilitates the communication between the host system and the guest system (gem5).
+The examples of how m5 is used could be found in the runscripts that we will describe later.
+m5 binary will be copied to the disk image, so that the guest could run m5 binary during the experiment.
+Therefore, m5 binary should be compiled before we build the disk image.
+
+**Note:** it's important to compile the m5 binary with -DM5_ADDR=0xFFFF0000 as is default in the SConscript.
```

Comment:
This note to me is a little big weird. You're saying M5_ADDR is set as a default in SCons, then saying if you don't compile with -DM5_ADDR you'll get an error, then continuing with an example where you don't use it.

I think what you want to say is "M5_ADDR" is set as 0xFFFF0000 by default in SConscript. Then explain what it's used for and the dangers in changing it.

Reply:
I don't really know the problem with changing M5_ADDR.

Reply:
I think Bobby was suggesting making an addition like

```markdown
**Note:** it's important to compile the m5 binary with -DM5_ADDR=0xFFFF0000 as is default in the SConscript.
This address is used by the guest binary to communicate with the simulator.
If you change the address in the guest binary, you also have to update the simulator to use the new address.
Additionally, when running in KVM, it is required that you use the *address* form of guest<->simulator communication and not the pseudo instruction form (i.e., using -DM5_ADDR is *required* when compiling a guest binary for which you want to run in KVM mode on gem5).
```